lib/ansible/module_utils/common/validation.py

```diff
@@ -195,23 +195,23 @@ def check_required_by(requirements, parameters, options_context=None):
     for (key, value) in requirements.items():
         if key not in parameters or parameters[key] is None:
             continue
-        result[key] = []
         # Support strings (single-item lists)
         if isinstance(value, string_types):
             value = [value]
         for required in value:
```

Comment:
Instead of collecting results and then throwing all but the first away, it would be much simpler to collect the missing values and raise the error immediately. Something like this:

```python
        if missing := [required for required in value if required not in parameters or parameters[required] is None]:
            msg = f"missing parameter(s) required by '{key}': {', '.join(missing)}"
            
            if options_context:
                msg = f"{msg} found in {' -> '.join(options_context)}"
                
            raise TypeError(to_native(msg))
```

Reply:
Thanks for this suggestion.